core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java

```diff
@@ -0,0 +1,583 @@
+    }
+
+    @Test
+    public void testMetaDataBefore() {
+        String triggerQuery = "UNWIND $createdNodes AS n SET n.label = labels(n)[0], n += $metaData";
+        String matchQuery = "MATCH (n:Bar) RETURN n";
+        testMetaData(triggerQuery, "before", matchQuery);
+    }
+
+    @Test
+    public void testMetaDataAfter() {
+        db.executeTransactionally("CREATE (n:Another)");
+        String triggerQuery = "UNWIND $createdNodes AS n MATCH (a:Another) SET a.label = labels(n)[0], a += $metaData";
+        String matchQuery = "MATCH (n:Another) RETURN n";
+        testMetaData(triggerQuery, "after", matchQuery);
+    }
+
+    private void testMetaData(String triggerQuery, String phase, String matchQuery) {
+        String name = "txinfo";
+        sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase:$phase})",
+                Map.of("name", name, "query", triggerQuery, "phase", phase));
+        awaitTriggerDiscovered(db, name, triggerQuery);
+
+        try (Transaction tx = db.beginTx()) {
+            KernelTransaction ktx = ((TransactionImpl)tx).kernelTransaction();
+            ktx.setMetaData(Collections.singletonMap("txMeta", "hello"));
+            tx.execute("CREATE (f:Bar)");
+            tx.commit();
+        }
+        
+        testCall(db, matchQuery, (row) -> {
+            final Node node = (Node) row.get("n");
+            assertEquals("Bar", node.getProperty("label"));
+            assertEquals("hello",  node.getProperty("txMeta"));
+        });
+    }
+
+    @Test
+    public void testPauseResult() {
+        String name = "pausedTest";
+        String query = "UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime";
+        sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'after'})", 
+                Map.of("name", name, "query", query));
+        awaitTriggerDiscovered(db, name, query);
+        testCall(sysDb, "CALL apoc.trigger.stop('neo4j', 'pausedTest')", (row) -> {
+            assertEquals("pausedTest", row.get("name"));
+            assertEquals(true, row.get("installed"));
+            assertEquals(true, row.get("paused"));
+        });
+    }
+
+    @Test
+    public void testPauseOnCallList() {
+        String name = "test";
+        String query = "UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime";
+        sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'after'})", 
+                Map.of("name", name, "query", query));
+        awaitTriggerDiscovered(db, name, query);
+        
+        sysDb.executeTransactionally("CALL apoc.trigger.stop('neo4j', 'test')");
+        testCallEventually(db, "CALL apoc.trigger.list()", (row) -> {
+            assertEquals("test", row.get("name"));
+            assertEquals(true, row.get("installed"));
+            assertEquals(true, row.get("paused"));
+        }, TIMEOUT);
+    }
+
```

Comment:
Could we add a test that checks the behaviour of the trigger is indeed pauses?

- Create Trigger "foo" which adds property "bar" to any node created
- Wait for "foo" to be installed
- Pause Trigger "foo"
- Wait for foo" to be paused
- Create node and make sure it doesn't have property "bar"

-